mods/tnt/init.lua

```diff
@@ -150,11 +150,13 @@ local function add_effects(pos, radius)
 	})
 end
 
-local function burn(pos)
+function tnt.burn(pos)
 	local name = minetest.get_node(pos).name
-	if name == "tnt:tnt" then
+	local group = minetest.get_node_group(name,tnt)
+	if group then
+		name = string.sub(name,5)
 		minetest.sound_play("tnt_ignite", {pos=pos})
-		minetest.set_node(pos, {name="tnt:tnt_burning"})
+		minetest.set_node(pos, {name="tnt:"..name.."_burning"})
```

Comment:
this assumes that all tnts have a mod namespace of `tnt`, which isn't a good assumption for an API.